python/shotgun_model/shotgun_model.py

```diff
@@ -669,19 +671,67 @@ def _finalize_item(self, item):
         """
         Called whenever an item is fully constructed, either because a shotgun query returned it
         or because it was loaded as part of a cache load from disk.
-        
-        .. note:: You can subclass this if you want to run post processing on 
+
+        .. note:: You can subclass this if you want to run post processing on
             the data as it is arriving. For example, if you are showing a list of
-            task statuses in a filter view, you may want to remember which 
+            task statuses in a filter view, you may want to remember which
             statuses a user had checked and unchecked the last time he was running
             the tool. By subclassing this UI you can easily apply those settings
-            before items appear in the UI. 
+            before items appear in the UI.
 
-        :param item: :class:`~PySide.QtGui.QStandardItem` that is about to be added to the model. 
+        :param item: :class:`~PySide.QtGui.QStandardItem` that is about to be added to the model.
             This has been primed with the standard settings that the ShotgunModel handles.
         """
         # the default implementation does nothing
 
+    def _set_tooltip(self, item, sg_item):
+        """
+        Called when an item is constructed for the first time.
```

Comment:
"For the first time" is confusing - it makes me think that this is only called when the data arrives from shotgun, but it is called every time a new model item is created right? If that's not the case, it should be - imagine a case where 3 different apps share the same cached sg data but want to display it in three different way, therefore needing three different tooltip display implementations!